### PR TITLE
Overhaul implementation

### PR DESCRIPTION
**Pull request by SgtCoDFish**
This is a large commit with a lot of changes:

- Change module name to match repo
- Fix several HTML errors
- Fix some error handling code in HTML / add specific error page
- Use a different name for certs, and require a key to fetch a cert
- Use a middleware to extract certs rather than duplicating logic across handlers
- Change QR code to include a link to the website rather than the actual PEM cert
- Add a mock mode to the print controller to allow local testing
- Allow download of private key PEM or a tar bundle
- Update footer to mention Venafi over Jetstack

### Example Front

front-636572742d6d616e61676572406578616d706c652e636f6d

### Example Back

back-636572742d6d616e61676572406578616d706c652e636f6d

### CODE REVIEW

**Comment by SgtCoDFish** [comment]
@maelvls @inteon this is now rebased and ready for a check if you're interested!

**Review by inteon — APPROVED** [review]
I don't have the time to manually test the code, but I reviewed the code and it looks good to me!

File: certificate.html [hunk 1]
```diff
@@ -19,20 +19,26 @@
       <img src="static/images/logo.svg" alt="Logo" width="220" height="191" />
       <h1 class="mainTitle">Print Your Certificate</h1>
       <div class="divider"></div>
+
+      {{if .Certificate }}
```
Comment (maelvls) [inline certificate.html]: This block one doesn't seem to end (probably due to the changes I made in #10, sorry about that!)


I get the following:

```
panic: template: certificate.html:132: unexpected EOF

goroutine 1 [running]:
html/template.Must(...)
	/opt/homebrew/Cellar/go/1.21.6/libexec/src/html/template/template.go:368
main.init()
	/Users/mvalais/code/print-your-cert/main.go:100 +0x2f8
exit status 2
```

Comment (SgtCoDFish) [reply certificate.html]: Oh damn, that was a refactoring mistake. I'll fix! Sorry

File: after-print.html [hunk 2]
```diff
@@ -26,6 +26,7 @@ <h1>Printing your certificate</h1>
       </div>
       <form method="GET" action="/certificate" class="form">
         <input name="certName" type="text" value="{{.CertName}}" hidden required />
+        <input name="certName" type="text" value="{{.FetchKey}}" hidden required />
```
Comment (maelvls) [inline after-print.html]:
```suggestion
        <input name="fetchKey" type="text" value="{{.FetchKey}}" hidden required />
```

Comment (SgtCoDFish) [reply after-print.html]: 🙏  thanks!

**Comment by SgtCoDFish** [comment]
@maelvls should be fixed now I think!

**Comment by maelvls** [comment]
I tested it with the following:


```bash
# First, I created the issuer from your PR:
kubectl apply -f cluster_issuer.yaml

# Then, in a shell session:
go run . --issuer root-print-your-cert-ca-issuer --namespace cert-manager --issuer-kind Issuer --listen 0.0.0.0:8080

# In another shell session:
./print-your-cert-controller
```

Then I went to the UI and submitted a cert.

I get stuck with the following in logs of the `go run` process:

> 2024/02/20 17:48:17 GET /certificate: 423: the requested certificate a52dbedbf10b1455adf033f313112c934896d7ed83d5d3a2e98328e94bb89831 in namespace cert-manager exists, but the Secret a52dbedbf10b1455adf033f313112c934896d7ed83d5d3a2e98328e94bb89831 does not.

I think the current PR only works if `--namespace` is set to `default`. It seems like I forgot to pass the namespace down:

https://github.com/cert-manager/print-your-cert/blob/591d11255d0c4abf3b3286e0fcd729750a8513fa/main.go#L320

→ opened a PR: https://github.com/SgtCoDFish/cert-manager-print-your-cert/pull/1

**Pull request merged**